src/components/utils/ZoomImage.tsx

```diff
@@ -0,0 +1,259 @@
+                        });
+                    }}
+                    className={classNames(
+                        props.className,
+                        zoomable ? styles.zoomImg : null,
+                        active ? styles.zoomImageActive : null,
+                    )}
+                />
+            )}
+        </>
+    );
+}
+
+function ZoomImageModal(props: { src: string; alt: string; onClose: () => void }) {
+    const { src, alt, onClose } = props;
+
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+
+    React.useEffect(() => {
+        const handleKeyDown = (event: KeyboardEvent) => {
+            if (event.key === 'Escape') {
+                onClose();
+            }
+        };
+
+        document.addEventListener('keydown', handleKeyDown);
+
+        return () => {
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    }, [onClose]);
+
+    React.useEffect(() => {
+        buttonRef.current?.focus();
+    }, []);
+
+    return (
+        <div
+            className={classNames(
+                styles.zoomModal,
+                tcls(
+                    'fixed',
+                    'inset-0',
+                    'z-50',
+                    'flex',
+                    'items-center',
+                    'justify-center',
+                    'bg-light',
+                    'dark:bg-dark',
+                    'p-8',
+                ),
+            )}
+            onClick={onClose}
+        >
+            <img
+                src={src}
+                alt={alt}
+                className={tcls(
+                    'max-w-full',
+                    'max-h-full',
+                    'object-contain',
+                    'bg-light',
+                    'dark:bg-dark',
+                )}
+            />
+
+            <button
+                ref={buttonRef}
+                className={tcls(
+                    'absolute',
+                    'top-5',
+                    'right-5',
+                    'flex',
+                    'flex-row',
+                    'items-center',
+                    'justify-center',
+                    'text-sm',
+                    'text-dark/6',
+                    'dark:text-light/5',
+                    'hover:text-primary',
+                    'p-4',
+                    'dark:text-light/5',
+                    'rounded-full',
+                    'bg-white',
+                    'dark:bg-dark/3',
+                    'shadow-sm',
+                    'hover:shadow-md',
+                    'border-slate-300',
+                    'dark:border-dark/2',
+                    'border',
+                )}
+                onClick={onClose}
+            >
+                <IconMinimize />
+            </button>
+        </div>
+    );
+}
+
+function startViewTransition(callback: () => void, onEnd?: () => void) {
+    // @ts-ignore
+    if (document.startViewTransition) {
+        // @ts-ignore
```

Comment:
We should get rid of `ts-ignore`

Reply:
This API is still experimental, it's fine